### PR TITLE
Added Modal component

### PR DESCRIPTION
**Pull request by BOTLANNER**
### Description:

Control to display modal content with optional header and footer content.

### Screenshots (if appropriate):
image

### All Submissions:

* [X] I have followed the [contribution guidelines](https://github.com/capitec/omni-components/blob/develop/CONTRIBUTING.md) for this project.
* [X] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.

### Changes to Core Features:

* [X] I have added an explanation of what my changes do and why I would like to include them.
* [X] I have written new tests for my core changes, as applicable.
* [X] I have successfully ran tests with my changes locally.
* [X] I have added/updated docs, as applicable.

### CODE REVIEW

File: src/modal/Modal.ts [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+ * ```
+ *
+ * @element omni-modal
+ *
+ * Registry of all properties defined by the component.
+ *
+ * @fires click-outside - Dispatched when a click or touch occurs outside the modal container.
+ *
+ * @slot header - Content to render inside the component header.
+ * @slot - Content to render inside the component body.
```
Comment (chromaticWaster) [inline src/modal/Modal.ts:37]: Add slot name

Comment (chromaticWaster) [reply src/modal/Modal.ts:37]: Noted that this is the default slot

**Review by chromaticWaster — APPROVED** [review]

File: src/modal/Modal.stories.ts
Comment (stefan505) [inline src/modal/Modal.stories.ts:1]: Consider modal close actions using buttons or icons or something.

File: src/modal/Modal.ts [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+                    }
+
+				}
+      `
+        ];
+    }
+
+    override render(): TemplateResult {
+        return html`
+        <dialog part="dialog" class="modal" role="dialog" aria-modal="true"
```
Comment (stefan505) [inline src/modal/Modal.ts:336]: Check indentation for template result functions, looks like tabs & spaces are mixed.

**Review by stefan505 — CHANGES_REQUESTED** [review]

**Review by stefan505 — APPROVED** [review]

**Pull request merged**